docs/src/lib/interfaces.md

```diff
@@ -116,3 +116,10 @@ A singleton is a special hyperrectangle consisting of only one point.
 ```@docs
 AbstractSingleton
 ```
+
+This interface defines the following functions:
+
+```@docs
+linear_map(M::AbstractMatrix, S::AbstractSingleton)
+linear_map(M::AbstractMatrix, H::AbstractHyperrectangle{Float64})
```

Comment:
Is there a reason why there is a numeric type for `AbstractHyperrectangle` and not for `AbstractSingleton`?

Reply:
In the former case i needed the numeric type `N`, in

```julia
...
    if dim(H) == 2
        T = VPolygon
    else
        T = VPolytope
    end
...
```
while in the latter, i didn't use `N`.

Reply:
it will compile to the same thing, but let's add the `N`.

on the other hand, annotating the output type is not obvious to me since it really depends on the concrete subtype of `AbstractSingleton`. hence, i would suggest not to write anything.

Reply:
Ah, you are right, I was only looking at the docs here. You cannot remove the `N` in the code.
But maybe you can remove it for the docs (not sure what happens). If it works, dropping the type for the docs would be good because it avoids the warning.